File: nilaiMatkulChecker.py
import time

print('---- Nilai Matkul Checker')
ans = 'Y'
while ans == 'Y' or ans == 'y':
    nama = input('masukkan nama anda\t: ')
    nim = input('masukkan nim anda\t: ')
    matkul = input('mata kuliah yang akan di cek\t: ')
    nilai = int(input('masukkan nilai dari matkul tsb\t: '))
    print('Data berhasil diinput!')

    if 85 <= nilai <= 100:
        nilaiHuruf = 'A'
    elif 75 <= nilai <= 85:
        nilaiHuruf = 'B'
    elif 65 <= nilai <= 75:
        nilaihuruf = 'C'
    elif nilai < 65:
        nilaiHuruf = 'D'

    time.sleep(1)
    print('sedang diproses')
    time.sleep(0.5)
    dots = '----'
    for dot in dots:
        print (dot)
        time.sleep(0.5)
    print('Nilai dari matkul anda masuk ke kategori ' + nilaiHuruf + '\n')
    time.sleep(1)
    ans = input('inginmencoba lagi? [Y/T] : ')

print('\nProgram berhenti')
    

File: nilaiMatkulChecker.c
#include <stdio.h>
#include <string.h>

int main(){
    char name[200], nim[20], matkul[100];
    char ans, nilaiHuruf;
    int nilai;

    printf("----Nilai Matkul Checker----");
    ans = 'Y';
    while (ans == 'Y' || ans == 'y'){
        printf("\nMasukkan nama Anda\t: ");
        scanf("%s", &name);
        printf("\nMasukkan NIM Anda\t: ");
        scanf("%s", &nim);
        printf("\nMata kuliah yang akan dicek\t: ");
        scanf("%s", &matkul);
        printf("\nMasukkan nilai dari matkul tsb\t: ");
        scanf("%d", &nilai);
        printf("\nData berhasil diinput!\n");

        if (nilai >= 85 && nilai <= 100){
            nilaiHuruf = 'A';
        } else if (nilai >= 75 && nilai < 85){
            nilaiHuruf = 'B';
        } else if (nilai >= 65 && nilai < 75){
            nilaiHuruf = 'C';
        } else if (nilai < 65){
            nilaiHuruf = 'D';
        }

        printf("Nilai dari %c\n", nilaiHuruf);
        printf("Ingin mencoba lagi? [Y/T] : ");
        scanf(" %s", &ans);

        fflush(stdin);
    }
    return 0;
}
